paella-matterhorn/javascript/modified-classes/videoloader-for-paella41.js

```diff
@@ -181,8 +181,61 @@ var MHVideoLoader = Class.create(paella.VideoLoader, {
     if (presenter) { this.streams.push(presenter); }
     if (presentation) { this.streams.push(presentation); }
 
-    // Callback
-    this.loadStatus = true;
-    onSuccess();      
+    // Get captions.
+    var captionURL = getCaptionURL();
+    if (captionURL) {
+      loadCaptions(captionURL, concludeLoading);
+    }
+    else {
+      setTimeout(concludeLoading, 0);
+    }
+
+    // Like MHCaptionsDataDelegate#read, except it stops when it finds the url
+    // and leaves loading the url up to other functions (like
+    // paella.captions.Caption#initialize and reload).
+    function getCaptionURL() {
+      var catalogs = paella.matterhorn.episode.mediapackage.metadata.catalog;
+      if (!(catalogs instanceof Array)) {
+        catalogs = [catalogs];
+      }
+      
+      var catalog;
+    
+      for (var i = 0; i < catalogs.length; ++i) {
+        catalog = catalogs[i];
+        if (catalog.type == 'captions/timedtext') {
+          break;
+        }
```

Comment:
This grabs any catalog, series or episode dublin core, if the captions catalog is not attached to the mediapackage. It should only set the catalog if the type is of 'captions/timedtext'. This accounts for the mysterious lingering caption icon.